client/python/examples/README.md

```diff
@@ -0,0 +1,265 @@
+
+### general.py
+
+[Link to File](https://github.com/G-Research/armada/tree/master/client/python/examples/general.py)
+
+A more fledged out version of `simple.py` where we create a queue only if it doesn't exist, and then create a jobset and job, and wait until the job succeeds or fails.
+
+#### Walkthrough
+
+```py
+def wait_for_job_event(client, event_stream, job_id: str, event_state: EventType)
+```
+
+We use this function to check the status of a job, return false if the job failed and otherwise wait for the event state we are interested in.
+
+**Please note** that this is shown for demonstration purposes only. Subscribing to events like this to watch individual events like this will not scale well.
+
+```py
+job_id = resp.job_response_items[0].job_id
+```
+
+We can use this code to get the job_id we are interested in from the event_stream.
+
+
+```py
+client.reprioritize_jobs(new_priority=2, queue=queue, job_set_id=job_set_id)
+
+# Needed to allow for the delay in the job_set being created
+time.sleep(2)
+```
+
+As the code documents, we need the sleep to wait for the job_set to be created, allowing for `client.get_job_events_stream` to not fail.
+
+```py
+try:
+    client.create_queue(name=queue, priority_factor=1)
+
+# Handle the error we expect to maybe occur
+except grpc.RpcError as e:
+    code = e.code()
+    if code == grpc.StatusCode.ALREADY_EXISTS:
+        print(f"Queue {queue} already exists")
+        client.update_queue(name=queue, priority_factor=1)
+    else:
+        raise e
+```
+
+For any grpc-related errors, more detail can be found by checking the grpc.RpcError object. In this case, we use it to ignore if a queue has already been created or not.
+
+### queues.py
+
+[Link to File](https://github.com/G-Research/armada/tree/master/client/python/examples/queues.py)
+
+A full example of creating a queue with all options.
+
+#### Walkthrough
+
+```py
+subject = Subject(type="Group", name="group1")
+permissions = Permissions(subjects=[subject], verbs=["cancel", "reprioritize"])
+```
+
+Permissions are used to control who can do what with the queue.
+
+```py
+resource_limits = {"cpu": 1.0, "memory": 1.0}
+```
+
+Resource limits are set similarly to [Kubernetes](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-requests-and-limits-of-pod-and-container)
+
+
+### cancelling.py
+
+[Link to File](https://github.com/G-Research/armada/tree/master/client/python/examples/cancelling.py)
+
+A full example of cancelling jobs, either with their job id, or
+the job-set id.
+
+#### Walkthrough
+
+```py
+job_id = resp1.job_response_items[0].job_id
+client.cancel_jobs(job_id=job_id)
+```
+
+To cancel a specific job, its job_id must be read from the `client.submit_jobs` response, and then it can be cancelled.
+
+```py
+client.cancel_jobs(queue=queue, job_set_id=job_set_id1)
+```
+
+You can also cancel all jobs on a job_set with its queue and job_set id.
```

Comment:
Show an example here also of that?  I think cancelling by jobset is the more common use case imo.